Automattic/jetpack
# Dashboard: Align module Protect data logic with Akismet

## Proposed changes:

Since we are tackling the Jetpack Stats Widget that consumes the API response of modules data, aligning module Protect logic with Akismet for the status check would improve the consistency.

* Align the consumed data from wp option `jetpack_protect_blocked_attempts` with the format of `akismet_spam_count`.
* Use only numeric and plugin status data for Akismet UI condition check.
* Align module Protect UI condition check logic with Akismet.

### Other information:


## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






1. Start on a test site with Brute force protection and Akismet Anti-spam enabled. The loading status of Brute force protection should be aligned with Akismet.

<img width="1073" alt="截圖 2023-04-21 上午12 54 11" src="https://user-images.githubusercontent.com/6869813/233444028-503167f3-9562-4013-8503-a835c53fb4cf.png">

2. Update the wp options of `jetpack_protect_blocked_attempts` and `akismet_spam_count` to `0`, and the status text of these two cards should be aligned.

<img width="1067" alt="截圖 2023-04-21 上午1 04 07" src="https://user-images.githubusercontent.com/6869813/233444912-3e75053e-4ffa-4f2f-89c1-474501a5c5e6.png">

3. Delete the wp options of `jetpack_protect_blocked_attempts` and `akismet_spam_count`, and the status text should be the same as `0`.

4. Set the wp options of `jetpack_protect_blocked_attempts` and `akismet_spam_count` to `1`, and the counts of two cards should be aligned.

<img width="1080" alt="截圖 2023-04-21 上午12 54 44" src="https://user-images.githubusercontent.com/6869813/233445495-021ce60f-fd70-4a14-9cb0-c8a762fc27a1.png">
